spec/models/provider_user_spec.rb

```diff
@@ -3,6 +3,38 @@
 require "rails_helper"
 
 describe ProviderUser do
+  describe "scopes" do
+    describe ".with_active_hei_providers" do
+      let(:active_hei_provider_user) { create(:provider_user, provider: create(:provider, :hei)) }
+      let(:inactive_hei_provider_user) { create(:provider_user, provider: create(:provider, :hei)) }
+      let(:unaccredited_hei_provider_user) { create(:provider_user, provider: create(:provider, :hei, :unaccredited)) }
+      let(:active_scitt_provider_user) { create(:provider_user, provider: create(:provider, :scitt)) }
+      let(:inactive_scitt_provider_user) { create(:provider_user, provider: create(:provider, :scitt)) }
+      let(:unaccredited_scitt_provider_user) { create(:provider_user, provider: create(:provider, :scitt, :unaccredited)) }
+
+      before do
+        inactive_hei_provider_user.provider.discard
+        inactive_scitt_provider_user.provider.discard
+
+        active_hei_provider_user
+        inactive_hei_provider_user
+        unaccredited_hei_provider_user
+        active_scitt_provider_user
+        inactive_scitt_provider_user
+        unaccredited_scitt_provider_user
+      end
+
+      it "returns provider user that have providers that are accredited and have a valid accreditation_id" do
+        expect(described_class.with_active_hei_providers).to include(active_hei_provider_user)
+        expect(described_class.with_active_hei_providers).not_to include(inactive_hei_provider_user)
+        expect(described_class.with_active_hei_providers).not_to include(unaccredited_hei_provider_user)
+        expect(described_class.with_active_hei_providers).not_to include(active_scitt_provider_user)
+        expect(described_class.with_active_hei_providers).not_to include(inactive_scitt_provider_user)
+        expect(described_class.with_active_hei_providers).not_to include(unaccredited_scitt_provider_user)
```

Comment:
```suggestion
        expect(described_class.with_active_hei_providers).to contain_exactly(active_hei_provider_user)
```

Reply:
Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/5191/commits/30dbaf8c4d947a180326f71a193d2e2902db014f